src/merkle/store/tests.rs

```diff
@@ -212,6 +212,7 @@ fn test_get_invalid_node() {
 }
 
 #[test]
+#[cfg(not(feature="concurrent"))]
 fn test_add_sparse_merkle_tree_one_level() -> Result<(), MerkleError> {
```

Comment:
I guess this test doesn't work because were are using `SmpleSmt` with depth that is not a multiple of 8, right?

If so, I think we have the following options:
1. Make concurrent construction work for trees of any depth.
2. Try to detect at compile time that we are trying to use a `SimpleSmt` with a depth that is not a multiple of 8.
3. Disable concurrent functionality for `SimpleSmt` (but still keep it for `Smt`).

The first option is not for this PR. Between 2nd and 3rd I'm leaning toward the 3rd as it would make the library a bit more intuitive (and then we can create an issue to solve it properly in the future).